marshmallow_utils/permissions.py

```diff
@@ -37,14 +38,17 @@ def _permissions_filter_load(self, data, **kwargs):
     @post_dump
     def _permissions_filter_dump(self, data, **kwargs):
         field_permission_check = self.context.get("field_permission_check")
-        # NOTE: we make a copy since we don't want to modify the data when
-        # deleting keys
-        # TODO (Alex): see if actually needed...
-        # data = deepcopy(data)
+        # Initialize "request" permissions cache
+        if not hasattr(request, "permissions"):
+            request.permissions_cache = dict()
         if field_permission_check:
             for k in self.field_dump_permissions:
                 if k in data:
                     action = self.field_dump_permissions[k] or self.default_dump_action
-                    if action and not field_permission_check(action):
+                    has_permission = request.permissions_cache.get(action)
```

Comment:
I think that the cache key should also be `data`, not only the `action`: in the same request, we might have to calculate the permission depending on some fields of the current data.

Reply:
The action is already using the data that is passed in the context in the schema (see [here](https://github.com/inveniosoftware/invenio-records-resources/blob/master/invenio_records_resources/services/records/schema.py#L73)). Once the action is evaluated by using that data from the context, I think that we could only store the outcome of the action, I have changed a bit the code, now we are not using the request object, we just store it in a variable as it's only used in that function. Here a screenshot to visualize how is caching each of the only 
![Screenshot from 2023-10-09 10-09-15](https://github.com/inveniosoftware/marshmallow-utils/assets/15194802/d35edf45-c770-4f7d-af43-e5d5a09d99f6)